File: dynabuffers-python/tests/usecase/Schema16Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema16Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema16.dbs"))

    def test_resolve_union_type_with_ordinal(self):
        message = {
            "stringVal": "abc",
            "intVal": 0,
            "byteVal": 1,
            "shortVal": 2,
            "longVal": 3,
            "floatVal": 4,
            "mapVal": {"name": "text"},
            "byteArrayVal": bytearray(b"text"),
            "refVal": {"name": "text"},
            ":type": 0
        }
        result = self.engine.deserialize(self.engine.serialize(message))
        self.assertEqual(result, message)

        self.assertEqual(result.get_string("stringVal"), "abc")
        self.assertEqual(result.get_int("intVal"), 0)
        self.assertEqual(result.get_byte("byteVal"), 1)
        self.assertEqual(result.get_short("shortVal"), 2)
        self.assertEqual(result.get_long("longVal"), 3)
        self.assertEqual(result.get_float("floatVal"), 4)
        self.assertEqual(result.get_map("mapVal"), {"name":"text"})
        self.assertEqual(result.get_bytearray("byteArrayVal"), bytearray(b"text"))
        self.assertEqual(result.get_ref("refVal"), {"name":"text"})

        with self.assertRaises(Exception) as ctx:
            result["unknown"]

        self.assertIsNotNone(ctx.exception)



if __name__ == "__main__":
    unittest.main()

File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema16Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema16Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema16Test::class.java.getResourceAsStream("/schema16.dbs"))
    }

    @Test
    fun dynabuffersMap() {
        val message = mapOf(
                "stringVal" to "abc",
                "intVal" to 0,
                "byteVal" to 1.toByte(),
                "shortVal" to 2.toShort(),
                "longVal" to 3.toLong(),
                "floatVal" to 4.toFloat(),
                "mapVal" to mapOf("a" to "b"),
                "byteArrayVal" to "text".toByteArray(),
                "refVal" to mapOf("name" to "text"),
                ":type" to 0
                )
        val result = engine.deserialize(engine.serialize(message))

        assertMap(message, result)
        result.getString("stringVal")
        result.getInt("intVal")
        result.getByte("byteVal")
        result.getShort("shortVal")
        result.getLong("longVal")
        result.getFloat("floatVal")
        result.getMap("mapVal")
        result.getByteArray("byteArrayVal")
        val ref = result.getRef("refVal")
        ref.getString("name")

        try {
            result["unknown"]
            Assertions.fail<String>("an exception is expected")
        } catch (e: Exception) {
        }
    }

    @Test
    fun dynabuffersMapWithEmptyList() {
        val message = mapOf(
                "stringVal" to "abc",
                "intVal" to 0,
                "byteVal" to 1.toByte(),
                "shortVal" to 2.toShort(),
                "longVal" to 3.toLong(),
                "floatVal" to 4.toFloat(),
                "mapVal" to mapOf("a" to emptyList<String>()),
                "byteArrayVal" to "text".toByteArray(),
                "refVal" to mapOf("name" to "text"),
                ":type" to 0
        )
        val result = engine.deserialize(engine.serialize(message))

        assertThat((result["mapVal"] as Map<String, Array<String>>)?.get("a")).isEmpty()
    }

}
